cg/services/orders/validation/rules/case_sample/rules.py

```diff
@@ -494,3 +497,19 @@ def validate_existing_samples_belong_to_collaboration(
                 )
                 errors.append(error)
     return errors
+
+
+def validate_sample_names_available(
+    order: OrderWithCases, store: Store, **kwargs
+) -> list[SampleNameAlreadyExistsError]:
+    """Validates that new sample names are not already used by the customer."""
+    errors: list[SampleNameAlreadyExistsError] = []
+    customer_entry_id: int = store.get_customer_by_internal_id(order.customer).id
+    for case_index, case in order.enumerated_new_cases:
+        for sample_index, sample in case.enumerated_new_samples:
```

Comment:
Very minor but this can be written on one line 
```suggestion
    for case_index, sample_index, sample in order.enumerated_new_samples:
```